crates/noirc_evaluator/src/ssa_refactor/ir/instruction/call.rs

```diff
@@ -0,0 +1,340 @@
+use std::rc::Rc;
+
+use acvm::{acir::BlackBoxFunc, BlackBoxResolutionError, FieldElement};
+use iter_extended::vecmap;
+use num_bigint::BigUint;
+
+use crate::ssa_refactor::ir::{
+    dfg::DataFlowGraph,
+    instruction::Intrinsic,
+    map::Id,
+    types::{NumericType, Type},
+    value::{Value, ValueId},
+};
+
+use super::{Endian, SimplifyResult};
+
+/// Try to simplify this call instruction. If the instruction can be simplified to a known value,
+/// that value is returned. Otherwise None is returned.
+pub(super) fn simplify_call(
+    func: ValueId,
+    arguments: &[ValueId],
+    dfg: &mut DataFlowGraph,
+) -> SimplifyResult {
+    use SimplifyResult::*;
+    let intrinsic = match &dfg[func] {
+        Value::Intrinsic(intrinsic) => *intrinsic,
+        _ => return None,
+    };
```

Comment:
```suggestion
    // Calls to generic Noir functions are simplified as part of the flattening and constant folding passes.
    // We only simplify intrinsic functions which are implemented as part of the compiler here.
    let intrinsic = match &dfg[func] {
        Value::Intrinsic(intrinsic) => *intrinsic,
        _ => return None,
    };
```

@jfecher Can you fact-check this first line on which passes are relevant?

Reply:
Looks like the comment was removed anyway, but I'll answer for completeness.

If by "simplified" we're referring to `Instruction::simplify` here, then this occurs in any pass that may (re)insert a call instruction - inlining, unrolling, flattening, and constant folding. However, I don't think this is too useful to point out since normal call instructions are never simplified in the first place. They are only ever completely removed during the inlining pass, after which we expect that all calls to Acir functions are removed and only brillig and intrinsic calls remain.